pkg/iam/iam.go

```diff
@@ -0,0 +1,409 @@
+	"ec2:DescribeTags",
+	"ec2:DescribeVolumes",
+	"ec2:DescribeVpcAttribute",
+	"ec2:DescribeVpcClassicLink",
+	"ec2:DescribeVpcClassicLinkDnsSupport",
+	"ec2:DescribeVpcs",
+	"ec2:DetachInternetGateway",
+	"ec2:DisassociateRouteTable",
+	"ec2:ModifyNetworkInterfaceAttribute",
+	"ec2:ModifyVpcAttribute",
+	"ec2:ReleaseAddress",
+	"ec2:ReplaceRouteTableAssociation",
+	"ec2:RevokeSecurityGroupEgress",
+	"ec2:RevokeSecurityGroupIngress",
+	"ec2:RunInstances",
+	"ec2:TerminateInstances",
+	"elasticloadbalancing:AddTags",
+	"elasticloadbalancing:ApplySecurityGroupsToLoadBalancer",
+	"elasticloadbalancing:AttachLoadBalancerToSubnets",
+	"elasticloadbalancing:ConfigureHealthCheck",
+	"elasticloadbalancing:CreateLoadBalancer",
+	"elasticloadbalancing:CreateLoadBalancerListeners",
+	"elasticloadbalancing:DeleteLoadBalancer",
+	"elasticloadbalancing:DeleteLoadBalancerListeners",
+	"elasticloadbalancing:DescribeLoadBalancerAttributes",
+	"elasticloadbalancing:DescribeLoadBalancers",
+	"elasticloadbalancing:DescribeTags",
+	"elasticloadbalancing:ModifyLoadBalancerAttributes",
+	"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+	"iam:AddRoleToInstanceProfile",
+	"iam:CreateInstanceProfile",
+	"iam:CreateRole",
+	"iam:DeleteInstanceProfile",
+	"iam:DeleteRole",
+	"iam:DeleteRolePolicy",
+	"iam:GetAccountAuthorizationDetails",
+	"iam:GetInstanceProfile",
+	"iam:GetRole",
+	"iam:GetUser",
+	"iam:ListAttachedGroupPolicies",
+	"iam:ListGroupsForUser",
+	"iam:ListGroupPolicies",
+	"iam:ListInstanceProfiles",
+	"iam:ListRolePolicies",
+	"iam:PassRole",
+	"iam:PutRolePolicy",
+	"iam:RemoveRoleFromInstanceProfile",
+	"route53:ChangeResourceRecordSets",
+	"route53:ChangeTagsForResource",
+	"route53:CreateHostedZone",
+	"route53:DeleteHostedZone",
+	"route53:GetChange",
+	"route53:GetHostedZone",
+	"route53:ListHostedZones",
+	"route53:ListResourceRecordSets",
+	"route53:ListTagsForResource",
+	"route53:UpdateHostedZoneComment",
+	"s3:CreateBucket",
+	"s3:DeleteBucket",
+	"s3:DeleteObject",
+	"s3:GetAccelerateConfiguration",
+	"s3:GetBucketCORS",
+	"s3:GetBucketLocation",
+	"s3:GetBucketLogging",
+	"s3:GetBucketRequestPayment",
+	"s3:GetBucketTagging",
+	"s3:GetBucketVersioning",
+	"s3:GetBucketWebsite",
+	"s3:GetEncryptionConfiguration",
+	"s3:GetLifecycleConfiguration",
+	"s3:GetObject",
+	"s3:GetObjectTagging",
+	"s3:GetReplicationConfiguration",
+	"s3:HeadBucket",
+	"s3:ListAllMyBuckets",
+	"s3:ListBucket",
+	"s3:PutBucketAcl",
+	"s3:PutBucketTagging",
+	"s3:PutObject",
+	"s3:PutObjectAcl",
+	"s3:PutObjectTagging",
+}
+
+// getAccountAuthDetails fetches lists of users, groups, and polices.
+func getAccountAuthDetails(svc *iam.IAM) AccountDetails {
+
+	var results AccountDetails
+
+	// Initial API call to get account auth details
+	input := &iam.GetAccountAuthorizationDetailsInput{}
+	resp, err := svc.GetAccountAuthorizationDetails(input)
+	if err != nil {
+		logrus.Fatalf("Error fetching account authorization details: %s", err.Error())
```

Comment:
Just a reminder this will execute `os.Exit(1)` and exit entirely. I believe that is what you are expecting, but bringing it up just in case.

Reply:
Yep, I do want it to exit. It's best to exit ASAP if the user doesn't have the right permissions. Because if we can't fetch authorization details, we won't know if the user has enough permission to install the cluster. They could hit errors further on in the install that don't make any sense to an end user (which is the problem this PR is trying to solve). Thanks for bringing it up.